common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -809,3 +810,123 @@ def test_perform_problem_reset_and_navigate(self):
+
+        self.dashboard_page = DashboardPage(self.browser)
+        self.progress_page = ProgressPage(self.browser, self.course_id)
+        self._setup_subsection()
+
+        # Auto-auth register for the course.
+        self._auto_auth(self.USERNAME, self.EMAIL, False)
+
+    def _setup_subsection(self):
+        """
+        Helper to set up a problem subsection as staff, then take
+        it as a student.
+        """
+        LogoutPage(self.browser).visit()
+        self._auto_auth("STAFF_TESTER", "staff101@example.com", True)
+        self.course_outline.visit()
+        self.course_outline.open_subsection_settings_dialog()
+
+        self.course_outline.select_advanced_tab()
+        self.course_outline.make_subsection_hidden_after_due_date()
+
+        LogoutPage(self.browser).visit()
+        self._auto_auth(self.USERNAME, self.EMAIL, False)
+        self.courseware_page.visit()
+
+        LogoutPage(self.browser).visit()
+
+    def _auto_auth(self, username, email, staff):
+        """
+        Logout and login with given credentials.
+        """
+        AutoAuthPage(self.browser, username=username, email=email,
+                     course_id=self.course_id, staff=staff).visit()
+
+    def create_multiple_choice_problem(self, problem_name):
+        """
+        Return the Multiple Choice Problem Descriptor, given the name of the problem.
+        """
+        factory = MultipleChoiceResponseXMLFactory()
+        xml_data = factory.build_xml(
+            question_text='The correct answer is Choice 2',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+
+        return XBlockFixtureDesc(
+            'problem',
+            problem_name,
+            data=xml_data,
+            metadata={'rerandomize': 'always'}
+        )
+
+    def test_subsecton_hidden_after_due_date(self):
+        """
+        Given that I am a staff member on the exam settings section
+        select advanced settings tab
+        When I Make the exam timed.
+        And I login as a verified student.
+        And visit the courseware as a verified student.
+        And I start the timed exam
+        Then I am taken to the exam with a timer bar showing
+        When I finish the exam
+        Then I see the exam submitted dialog in place of the exam
+        When I log back into studio as a staff member
+        And change the problem's due date to be in the past
+        And log back in as the original verified student
+        Then I see the exam or message in accordance with the hide_after_due setting
+        """
+        LogoutPage(self.browser).visit()
+        self._auto_auth(self.USERNAME, self.EMAIL, False)
+        self.courseware_page.visit()
+        self.assertFalse(self.courseware_page.has_past_due_date_message())
+
+        self.progress_page.visit()
+        self.assertEqual(self.progress_page.scores('Test Section 1', 'Test Subsection 1'), [(0,1)])
```

Comment:
This assertion is what's failing when I run locally - the problem is not appearing on the progress page at all for some reason

<img width="946" alt="screen shot 2016-07-22 at 3 03 45 pm" src="https://cloud.githubusercontent.com/assets/7373924/17068203/94b7cb38-501d-11e6-935b-18fda3fcc921.png">